core/src/main/java/com/facebook/testing/screenshot/internal/MetadataRecorder.java

```diff
@@ -0,0 +1,168 @@
+      }
+      allScreenshotsMetadata.add(mCurrentScreenshotMetadata);
+    }
+
+    public ScreenshotMetadataRecorder withDescription(String description) {
+      mCurrentScreenshotMetadata.description = description;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withName(String name) {
+      mCurrentScreenshotMetadata.name = name;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withTestClass(String testClass) {
+      mCurrentScreenshotMetadata.testClass = testClass;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withTestName(String testName) {
+      mCurrentScreenshotMetadata.testName = testName;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withTileWidth(int width) {
+      mCurrentScreenshotMetadata.tileWidth = width;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withTileHeight(int height) {
+      mCurrentScreenshotMetadata.tileHeight = height;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withViewHierarchy(String viewHierarchyFilename) {
+      mCurrentScreenshotMetadata.viewHierarchy = viewHierarchyFilename;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withAxIssues(String axIssuesFilename) {
+      mCurrentScreenshotMetadata.axIssues = axIssuesFilename;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withExtras(Map<String, String> extras) {
+      mCurrentScreenshotMetadata.extras = new HashMap<>(extras);
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withError(String error) {
+      mCurrentScreenshotMetadata.error = error;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withGroup(String group) {
+      mCurrentScreenshotMetadata.group = group;
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withAbsoluteFileName(String absolutePath) {
+      if (mCurrentScreenshotMetadata.absoluteFilesNames == null) {
+        mCurrentScreenshotMetadata.absoluteFilesNames = new ArrayList<>();
+      }
+      mCurrentScreenshotMetadata.absoluteFilesNames.add(absolutePath);
+      return this;
+    }
+
+    public ScreenshotMetadataRecorder withRelativeFileName(String relativePath) {
+      if (mCurrentScreenshotMetadata.relativeFileNames == null) {
+        mCurrentScreenshotMetadata.relativeFileNames = new ArrayList<>();
+      }
+      mCurrentScreenshotMetadata.relativeFileNames.add(relativePath);
+      return this;
+    }
+  }
+
+  File getMetadataFile() {
+    return new File(mDir, "metadata.json");
+  }
+
+  private List<ScreenshotMetadata> getOrReadMetadata() throws IOException {
+    if (mMetadata == null) {
+
+      File metadataFile = getMetadataFile();
+      if (metadataFile.exists()) {
+        Gson gson = new Gson();
+        JsonReader jsonReader = new JsonReader(new FileReader(getMetadataFile()));
+        mMetadata = gson.fromJson(jsonReader, new TypeToken<List<Map<String, Object>>>() {
```

Comment:
Why isnt this type token for the type `List<ScreenshotMetadata>` so that it matches the type of `mMetadata` ?

Reply:
fixed, thanks